test/functional/search_controller_test.rb

```diff
@@ -155,6 +155,58 @@ class SearchControllerTest < ActionController::TestCase
     assert_select 'a[href="#test-123"]', count: 0
   end
 
+
+  test 'hyphen and unicode hyphen are treated equally in search' do
```

Comment:
you realise this isn't testing the search, as it is stubbed out ?
but still useful I think for checking the flash notice text contains the original text.